app/controllers/providers/omniauth_callbacks_controller.rb

```diff
@@ -23,5 +24,18 @@ def after_omniauth_failure_path_for(_)
     def auth_hash
       request.env['omniauth.auth']
     end
+
+    def check_provider_is_allowed
```

Comment:
'allowed' feels strange to me for some reason. Although I guess is clear. 
Just wondering if you'd consider 'authorised' or 'enrolled'

Reply:
The provider is technically authorised already at this point, as there is an additional (Devise) filter that handles the SAML response from Portal. That's why I didn't named it authorised. Just to avoid confusion!

Enrolment is a good one, I will be using that one when the rules based on office code or email are implemented 👍

Reply:
Great ok, thanks for explaining!